Controls/DataControls/DataControls.UWP/ListView/View/Controls/RadListViewItem.IDragDropElement.cs

```diff
@@ -265,8 +266,51 @@ void IDragDropElement.OnDragDropComplete(DragCompleteContext context)
             {
                 this.FinalizeReorder(context);
 
-                object destinationDataItem = this.GetDestinationDataItem(data.CurrentSourceReorderIndex);
-                bool isExecuted = this.ListView.commandService.ExecuteCommand(CommandId.ItemReorderComplete, new ItemReorderCompleteContext(data.Data, destinationDataItem, this));
+                var dataItem = data.Data;
+                var destinationDataItem = this.GetDestinationDataItem(data.CurrentSourceReorderIndex);
+
+                IDataGroup dataGroup = null;
+                IDataGroup destinationDataGroup = null;
+
+                if (this.listView.GroupDescriptors.Count > 0)
+                {
+                    dataGroup = this.listView.Model.FindItemParentGroup(dataItem);
+                    destinationDataGroup = this.listView.Model.FindItemParentGroup(destinationDataItem);
+                }
+
+                ItemPlacement placement;
+
+                if (data.InitialSourceIndex < data.CurrentSourceReorderIndex)
+                {
+                    placement = ItemPlacement.After;
+                }
+                else
+                {
+                    placement = ItemPlacement.Before;
+                }
+
+                var commandContext = new ItemReorderCompleteContext(dataItem, dataGroup, destinationDataItem, destinationDataGroup, placement);
+                var isExecuted = this.ListView.commandService.ExecuteCommand(CommandId.ItemReorderComplete, commandContext);
+
+                if (isExecuted)
+                {
+                    // TODO: Data provider does not handle well reordering of items in groups.
+                    // Remove this workaround once we fix the reordering in the data provider.
+                    if (this.listView.GroupDescriptors.Count > 0)
+                    {
+                        var dataProvider = this.listView.Model.CurrentDataProvider as DataProviderBase;
```

Comment:
The if body could be simplified by calling: `this.listView.updateService.RegisterUpdate((int)UpdateFlags.AffectsData);`